sycl/tools/lspi.cpp

```diff
@@ -0,0 +1,167 @@
+//==----------- lspi.cpp ---------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// The "lspi" utility lists all platforms/devices discovered by PI similar to
+// how lscl prints this for OpenCL devices. It can probably be eventually merged
+// with the more complex "check-sycl" utility.
+//
+// There are two types of output:
+//   concise (default) and
+//   verbose (enabled with any argument).
+//
+// In verbose mode it also prints, which devices would be chose  by various SYCL
+// device selectors.
+//
+#include <CL/sycl.hpp>
+
+#include <cstdlib>
+#include <iostream>
+#include <stdlib.h>
+
+using namespace cl::sycl;
+using namespace std;
```

Comment:
Is this really needed? 
`std` is used in various places down the line anyway 
Its better to restrict the `using namespace` down to function scopes, or at least restrict to the types aliased, e.g.

```
using std::cout;
using  std::string;
```

Reply:
OK

Reply:
Ahem...
I am unsure it will pass a Clang/LLVM community review.